### PR TITLE
Feature/niad 425 sds api fhir compliant

### PR DESCRIPTION
**Pull request by jaroslaw-wojciechowski**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: docker/sds-api/spine-directory-service/sds/request/content_type_validator.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+from comms.http_headers import HttpHeaders
+from tornado.httputil import HTTPHeaders
+from utilities import integration_adaptors_logger as log
+
+logger = log.IntegrationAdaptorsLogger(__name__)
+valid_content_types = ['application/json', 'application/fhir+json', 'application/fhir+xml']
+
+
+def get_valid_content_type(headers: HTTPHeaders):
+    content_type = headers.get(HttpHeaders.CONTENT_TYPE, None)
```
Comment (petervdm) [inline docker/sds-api/spine-directory-service/sds/request/content_type_validator.py:11]: Maybe just default this to `application/json` then you don't need the first if?

File: docker/sds-api/spine-directory-service/sds/request/content_type_validator.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    if content_type is None:
+        return 'application/json'
+    elif content_type in valid_content_types:
+        return content_type
+    else:
+        logger.info("Invalid content-type in request")
+        raise tornado.web.HTTPError(400, 'Invalid content-type in request',
+                                    reason=f'Invalid content-type in request')
```
Comment (Alicja-Kainos) [inline docker/sds-api/spine-directory-service/sds/request/content_type_validator.py:19]: Is there a reason to have the same text in error message and reason? Should one of it nbe more specific and i.e. print out the content_type?

File: docker/sds-api/spine-directory-service/sds/request/content_type_validator.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+from request.http_headers import HttpHeaders
+from utilities import integration_adaptors_logger as log
+
+logger = log.IntegrationAdaptorsLogger(__name__)
+valid_accept_types = ['application/json', 'application/fhir+json', 'application/fhir+xml']
+
+
+def get_valid_accept_type(headers: HTTPHeaders):
+    accept_type = headers.get(HttpHeaders.ACCEPT, 'application/json')
+    if accept_type == '*/*':
```
Comment (jaroslaw-wojciechowski) [inline docker/sds-api/spine-directory-service/sds/request/content_type_validator.py:13]: That's the default for curl if Accept header is missing

File: docker/sds-api/spine-directory-service/sds/request/content_type_validator.py (hunk 2 above)
Comment (jaroslaw-wojciechowski) [reply docker/sds-api/spine-directory-service/sds/request/content_type_validator.py]: updated

File: docker/sds-api/spine-directory-service/sds/request/content_type_validator.py (hunk 3 above)
Comment (petervdm) [reply docker/sds-api/spine-directory-service/sds/request/content_type_validator.py:13]: Never too old to learn! :)

**Pull request merged**